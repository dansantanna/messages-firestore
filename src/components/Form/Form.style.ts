import { styled } from "@stitches/react";

export const Container = styled("form", {
  display: "flex",
  flexDirection: "column",
  marginBottom: 24,
  padding: "0 16px",
  h2: { marginBottom: 8 },
  button: { marginTop: 16 },
});
