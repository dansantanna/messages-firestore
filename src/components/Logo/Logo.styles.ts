import { styled } from "utils/stitches.config";

export const Image = styled("img", {
  height: "10vh",
  minHeight: 50,
  marginTop: 57,
  marginBottom: 31,
  alignSelf: "center",
});
