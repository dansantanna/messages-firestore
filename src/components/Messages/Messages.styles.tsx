import { styled } from "utils/stitches.config";

export const Container = styled("div", {
  backgroundColor: "$backgroundSecundary",
  padding: "22px 16px",
  display: "flex",
  gap: "16px",
  flexDirection: "column",
  flex: 1,
});
