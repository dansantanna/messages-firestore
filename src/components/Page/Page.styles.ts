import { styled } from "utils/stitches.config";

export const Container = styled("div", {
  display: "flex",
  flexDirection: "column",
  height: "100vh",
});
