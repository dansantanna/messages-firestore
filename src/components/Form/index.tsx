import Button from "components/Button";
import Heading from "components/Heading";
import useMessages from "hooks/useMessages";
import { useState } from "react";
import { Container } from "./Form.style";
import Input from "components/Input";

const Form = () => {
  const [messageInput, setMessageInput] = useState("");
  const { addMessage } = useMessages();

  const handleSubmit = () => {
    addMessage(messageInput);
    setMessageInput("");
  };

  return (
    <Container onSubmit={(evt) => evt.preventDefault()}>
      <Heading>Digite um texto abaixo</Heading>
      <Input
        id="message"
        placeholder="Insira sua mensagem*"
        value={messageInput}
        onChange={(evt) => setMessageInput(evt.currentTarget.value)}
      />
      <Button disabled={!messageInput} onClick={handleSubmit}>
        Enviar
      </Button>
    </Container>
  );
};

export default Form;
