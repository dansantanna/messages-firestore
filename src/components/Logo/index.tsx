import logo from "assets/logo.svg";
import { Image } from "./Logo.styles";

const Logo = () => {
  return <Image src={logo} />;
};

export default Logo;
