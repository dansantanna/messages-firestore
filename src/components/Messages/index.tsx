import Heading from "components/Heading";
import useMessages from "hooks/useMessages";
import { Container } from "./Messages.styles";
import Message from "components/Message";

const Messages = () => {
  const { messages } = useMessages();

  return (
    <Container>
      <Heading>Mensagens enviadas</Heading>
      {messages.map((message) => (
        <Message key={message.id} {...message} />
      ))}
    </Container>
  );
};

export default Messages;
