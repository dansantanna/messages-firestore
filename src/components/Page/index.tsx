import Messages from "components/Messages";
import Form from "components/Form";
import Logo from "components/Logo";
import { Container } from "./Page.styles";

const Page = () => {
  return (
    <Container>
      <Logo />
      <Form />
      <Messages />
    </Container>
  );
};

export default Page;
