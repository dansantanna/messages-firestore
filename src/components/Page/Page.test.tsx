import { render, screen } from "@testing-library/react";
import Page from ".";
import Logo from "components/Logo";
import Form from "components/Form";
import Messages from "components/Messages";

// Mocking the child components
jest.mock("components/Logo");
jest.mock("components/Form");
jest.mock("components/Messages");

describe("Page Component", () => {
  beforeEach(() => {
    // Mock implementations for the child components
    (Logo as jest.Mock).mockImplementation(() => <div>Mocked Logo</div>);
    (Form as jest.Mock).mockImplementation(() => <div>Mocked Form</div>);
    (Messages as jest.Mock).mockImplementation(() => (
      <div>Mocked Messages</div>
    ));
  });

  it("renders the Logo component", () => {
    render(<Page />);
    expect(screen.getByText("Mocked Logo")).toBeInTheDocument();
  });

  it("renders the Form component", () => {
    render(<Page />);
    expect(screen.getByText("Mocked Form")).toBeInTheDocument();
  });

  it("renders the Messages component", () => {
    render(<Page />);
    expect(screen.getByText("Mocked Messages")).toBeInTheDocument();
  });

  it("renders all components inside the Container", () => {
    const { container } = render(<Page />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
