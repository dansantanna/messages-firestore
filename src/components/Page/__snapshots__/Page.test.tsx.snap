// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Component renders all components inside the Container 1`] = `
<div
  class="c-PSqgi"
>
  <div>
    Mocked Logo
  </div>
  <div>
    Mocked Form
  </div>
  <div>
    Mocked Messages
  </div>
</div>
`;
